src/orquestra/sdk/_base/_graphs.py

```diff
@@ -50,15 +55,15 @@ def topological_sort(graph_to_sort: Graph) -> t.List[Node]:
     S = _root_nodes(graph)
 
     while S:
-        n = S.pop()
+        n = S.popitem()[0]
         L.append(n)
         n_deps = list(graph.get(n, []))
         for m in n_deps:
-            graph[n].remove(m)
-            inverted_graph.get(m, set()).remove(n)
+            graph[n].pop(m)
+            inverted_graph.get(m, dict()).pop(n)
 
             if not inverted_graph.get(m):
-                S.add(m)
+                S[m] = None
```

Comment:
**suggestion**: after this PR we'll be dealing with `dict`s with `None` as values. `d.get("k")` will return `None` even if the value's there. Changing the condition into an explicit form might be less bug-prone.

```
if m not in inverted_graph:
```
or
```
if len(inverted_graph.get(m, dict())) == 0:
```

Reply:
inverted_graph is dict of {node, dict} (was {node, set} before my changes). This checks if inverted_graph(m) == {}. "None" object ive added to internal dict has nothing to do with this line of code.